pkg/node/depth.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package node
+
+import (
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/topology"
+)
+
+type batchStoreDepther struct {
+	rs   postage.ReserveStateGetter
+	base swarm.Address
+}
+
+func (d *batchStoreDepther) NeighborhoodDepth() uint8 {
+	return d.rs.GetReserveState().StorageRadius
```

Comment:
This is not correct. 
The depth for a full node is minimum of connection depth and storage radius

Reply:
What we actually need is for pull sync to use storage radius instead of depth, not to always show storage radius as depth, exactly for the aforementioned reason, e.g. in protocols it would lead to a too narrow understanding of neighborhood, no immediate replication, etc.

Reply:
> pointed out that this is not possible because at a high storage radius, there may no peers in the neighborhood.

This is why the depthmonitor has the decreasing mechanism to find the optimal radius.  
The storage radius is zero for light nodes, so using the minimum of both for light nodes, won't work. This is why for full nodes we should use the storage radius directly, and for light nodes, we use the connection depth.

> pointed out that this is not possible because at a high storage radius, there may no peers in the neighborhood.

The reverse can also be true where the connection depth is too low where at this depth there's too much content to put in our reserve.